skimage/feature/peak.py

```diff
@@ -11,8 +11,8 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     Peaks are the local maxima in a region of `2 * min_distance + 1`
     (i.e. peaks are separated by at least `min_distance`).
 
-    If peaks are flat (i.e. multiple adjacent pixels have identical
-    intensities), the coordinates of all such pixels are returned.
+    If multiple peaks in the specified region have identical intensities,
+    the coordinates of all such pixels are returned.
```

Comment:
I don't really understand this formulation--can you explain what this means?

Reply:
This is probably the wrong place to discuss this (given the PR is closed), but the thread is most closely related to what I was thinking about. So adding this here for now. Happy to move it to a more appropriate active thread.

It's true that it is really hard to break ties between very close peaks. Personally can't think of a good way to do this, but maybe someone has already thought about this since.

If we instead focus on the downstream usage of this function in generating seeds for segmentation, we could try to unify neighboring peaks within a specified distance using a morphological dilation possibly followed by a morphological erosion (of smaller size). This would generate one seed (or group of connected seeds) in the middle of each region, which would get only one label.

Reply:
@jakirkham LOL Yes it's an interesting choice of forum (on some code comments, too, rather than the main discussion thread). No matter. =)

The operations you suggest, to me seem better suited for preprocessing before this function, rather than being part of the function. At any rate, this whole discussion might be better left until *after* #3022 is merged. I suspect we will have major algorithmic changes to this function that will affect the range of possible/desirable solutions.